scripts/replay_commits.py

```diff
@@ -241,21 +242,155 @@ def replay_commit(repo: str, original_sha: str, parent_sha: str) -> str | None:
+
+2. Write the PR body to: {body_path}
+   - Start with a brief summary (1-2 sentences)
+   - Include a bullet list of changes
+   - If there's an issue number, include "Closes #N" or "Fixes #N"
+   - Use markdown formatting
+
+Write ONLY to these files. Do not output anything else."""
+
+    try:
+        subprocess.run(
+            ["opencode", "run", prompt],
+            check=True,
+            capture_output=True,
+            text=True,
+            timeout=120,
+        )
 
+        title = Path(title_path).read_text().strip()
+        body = Path(body_path).read_text().strip()
+
+        # Validate title length
+        if len(title) > 72:
+            title = title[:69] + "..."
+
+        return title, body
+
+    except (
+        subprocess.CalledProcessError,
+        subprocess.TimeoutExpired,
+        FileNotFoundError,
+    ):
+        # Fallback if opencode fails
+        return generate_fallback_pr_content(commits, config)
+    finally:
+        # Clean up temp files
+        for path in [title_path, body_path]:
+            try:
+                Path(path).unlink()
+            except OSError:
+                pass
+
+
+def generate_fallback_pr_content(commits: list[str], config: Config) -> tuple[str, str]:
+    """Generate PR content without AI (fallback)."""
     if config.issue_number and config.issue_title:
         title = config.issue_title
-        body = f"Closes #{config.issue_number}"
-    elif first_commit_subject:
-        title = first_commit_subject
-        body = ""
+    elif commits:
+        title = get_commit_subject(commits[0])
+    else:
+        title = f"Changes from {config.current_branch}"
+
+    body_parts = []
+    if len(commits) > 1:
+        body_parts.append("## Changes\n")
+        for sha in commits:
+            subject = get_commit_subject(sha)
+            body_parts.append(f"- {subject}")
+        body_parts.append("")
+
+    return title, "\n".join(body_parts)
+
+
+def create_pull_request(config: Config, head_sha: str) -> str:
+    commits = get_commits(config.start_sha)
+
+    if len(commits) == 1:
+        title = get_commit_subject(commits[0])
+        body = get_commit_body(commits[0])
+        if config.issue_number and not body_has_issue_reference(
+            body, config.issue_number
+        ):
+            if body:
+                body += f"\n\nCloses #{config.issue_number}"
+            else:
+                body = f"Closes #{config.issue_number}"
+    elif len(commits) > 1:
+        print("Generating PR title and body from commits...")
+        title, body = generate_pr_content_with_ai(commits, config)
+        if config.issue_number and not body_has_issue_reference(
+            body, config.issue_number
+        ):
+            body += f"\n\nCloses #{config.issue_number}"
     else:
         title = f"Changes from {config.current_branch}"
         body = ""
+        if config.issue_number:
+            body = f"Closes #{config.issue_number}"
```

Comment:
@dobbyphus could we not just use this switch to set the body, then have just one check at the end for if config.issue_number and not body_has_issue_reference? that would be easier to follow.